src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -43,19 +43,96 @@
 vprintln(args...) = nothing
 
 using LinearAlgebra
-using BlockArrays # used for the `mortar` command which makes block matrices
+import Base: *
+import Base.size, Base.adjoint, Base.Array
 
-# when X or Y are BlockArrays, this makes the return value be a proper array (not a BlockArray)
-function array_mul(X::AbstractArray{T}, Y) where {T}
-    Z = Array{T}(undef, size(X, 1), size(Y, 2))
-    mul!(Z, X, Y)
+include("../workarounds/gpu_arrays.jl")
+
+# For now, BlockMatrix can store arrays of different types (for example, an element 
+# of type views and one of type Matrix). Maybe for performance issues it should only
+# store arrays of the same type?
+
+struct BlockMatrix{T <: Number, D <: Tuple} <: AbstractMatrix{T}
```

Comment:
In particular it would be nice if it was more clear that it is column blocks, but I don't find a good name, so just make it clear from a comment that it is a horizontal concatenation of blocks

Reply:
note that these are relatively simple wrappers only supporting some multiplication routines, and in particular operations with them materialize to plain arrays

Reply:
also explain relationship to BlockArrays (it's a lightweight subset of functionality, it materializes to plain arrays, it supports GPU)

Reply:
Re name : LazyHcat? (because it's basically equivalent to `hcat` but better optimized)

Reply:
I think LazyHcat is not too bad: it's clearer to have "hcat" than to just have the generic "block" which doesn't give a lot of information. And we probably never will have to implement a true block-matrix structure for LOBPCG.